mentat/code_context.py

```diff
@@ -74,7 +76,47 @@ def _get_all_features(
             )
             all_features.append(_feature)
 
-    return all_features
+    return sorted(all_features, key=lambda f: f.path.relative_to(git_root))
+
+
+def _merge_features(a: list[CodeFeature], b: list[CodeFeature]) -> list[CodeFeature]:
+    output = list[CodeFeature]()
+    for feature in a + b:
+        matching = [f for f in output if f.path == feature.path]
+        intersects = (
+            False
+            if not matching
+            else any(
+                i_feat.intersects(i_match)
+                for i_feat in feature.intervals
+                for i_match in matching[0].intervals
+            )
+        )
+        if not matching or not intersects:
+            output.append(feature)
+            continue
+
+        # Replace matching feature's intervals with a single interval covering
+        # the entire range of both features.
+        # TODO: Our features currently support multiple intervals so this step
+        # could add additional lines, but it would be better to make all features
+        # single-interval than build-out this.
+        feature_start = min(feature.intervals, key=lambda i: i.start)
+        feature_end = max(feature.intervals, key=lambda i: i.end)
+        matching_start = min(matching[0].intervals, key=lambda i: i.start)
+        matching_end = max(matching[0].intervals, key=lambda i: i.end)
+        matching[0].intervals = [
+            Interval(
+                min(feature_start.start, matching_start.start),
+                max(feature_end.end, matching_end.end),
+            )
+        ]
+        if feature.diff and not matching[0].diff:
+            matching[0].diff = feature.diff
+        if feature.user_included and not matching[0].user_included:
+            matching[0].user_included = feature.user_included
```

Comment:
I'm confused. Does this matching feature ever get added to output?

Reply:
It should - that was a mistake